src/languages/yaml.js

```diff
@@ -86,15 +86,21 @@ function(hljs) {
       },
       { // array listing
         className: 'bullet',
-        begin: '^ *-',
+      // TODO: remove |$ and |^ hacks when we have proper look-ahead support
+      begin: '((?<= )|^)?\\-(?=[ ]|$)',
         relevance: 0
       },
       hljs.HASH_COMMENT_MODE,
       {
         beginKeywords: LITERALS,
         keywords: {literal: LITERALS}
       },
-      hljs.C_NUMBER_MODE,
+      // numbers are any valid C-style number that
+      // sit isolated from other words
+      {
+        className: 'number',
+        begin: hljs.C_NUMBER_RE + '\\b'
```

Comment:
Do we not need a boundary before also?  The last match might not always have been a \b.

Reply:
I _think_ the first boundary is included in the `C_NUMBER_RE` but I'll check and confirm.

Reply:
I didn't see it, but I only glanced quickly.

Reply:
Yeah it's in there, but isn't in front of the `-`...unsure what the impact of that would be:
`'(-?)(\\b0[xX][a-fA-F0-9]+|(\\b\\d+(\\.\\d*)?|\\.\\d+)([eE][-+]?\\d+)?)'`

I'll add a few tests to check it's working as expected.

Reply:
Ah, no that's prolly ok. :-)